File: Python/day2.py
# Part 1
lines = [line.strip() for line in open("aoc_2.txt")]
points = {"X": 1, "Y": 2, "Z": 3}
score = {"A": {"X": 3, "Y": 6, "Z": 0}, 
        "B": {"X": 0, "Y": 3, "Z": 6}, 
        "C": {"X": 6, "Y": 0, "Z": 3}}
total = 0
for line in lines:
    in1, in2 = line.split()
    total += int(points[in2]) + int(score[in1][in2])
print(total)

# Part 2
new_move = {"A": {"X": "Z", "Y": "X", "Z": "Y"},
            "B": {"X": "X", "Y": "Y", "Z": "Z"},
            "C": {"X": "Y", "Y": "Z", "Z": "X"}}
total = 0
for line in lines:
    in1, in2 = line.split()
    total += int(points[new_move[in1][in2]]) + int(score[in1][new_move[in1][in2]])
print(total)

File: R/day2.R
aoc_2 <- read.table("~/Desktop/aoc_2.txt")
aoc_2$V2 <- ifelse(aoc_2$V2 == "X", "A", ifelse(aoc_2$V2 == "Y", "B", "C"))

rps <- function(input1, input2) {
  points <- ifelse(input2 == "A", 1, ifelse(input2 == "B", 2, 3))
  score <- ifelse(input1 == input2, 3, 
                  ifelse(input2 == "A" & input1 == "C" |
                           input2 == "C" & input1 == "B" |
                            input2 == "B" & input1 == "A", 6, 0))
  return(points + score)
}
sum(rps(aoc_2$V1, aoc_2$V2))

move <- function(input1, input2) {
  if (input2 == "B") {
    return(input1)
  }
  return(ifelse(input1 == "A" & input2 == "A" | input1 == "B" & input2 == "C", "C", 
         ifelse(input1 == "A" & input2 == "C" | input1 == "C" & input2 == "A", "B", "A")))
}
aoc_2$new_move <- apply(aoc_2, 1, function(row) move(row[1], row[2]))
sum(apply(aoc_2, 1, function(row) rps(row[1], row[3])))